src/client/app/effects/snippet.ts

```diff
@@ -288,6 +288,13 @@ export class SnippetEffects {
             case Snippet.ImportType.SAMPLE:
             case Snippet.ImportType.URL:
             case Snippet.ImportType.GIST:
+                let viewUrlMatch = new RegExp(`^${environment.current.config.editorUrl}\/#\/view`);
+                if (type === Snippet.ImportType.URL && viewUrlMatch.test(data)) {
+                    /* If importing from a view-mode URL, simply redirect the user */
```

Comment:
I don't get it.  Don't we want to import rather than navigating the user away?..